backend/open_webui/utils/tools.py

```
@@ -183,37 +275,323 @@ def function_to_pydantic_model(func: Callable) -> type[BaseModel]:
+                # Check if URL ends with .yaml or .yml to determine format
+                if url.lower().endswith((".yaml", ".yml")):
+                    text_content = await response.text()
+                    res = yaml.safe_load(text_content)
+                else:
+                    res = await response.json()
+    except Exception as err:
+        log.exception(f"Could not fetch tool server spec from {url}")
+        if isinstance(err, dict) and "detail" in err:
+            error = err["detail"]
+        else:
+            error = str(err)
```

Ruff (0.12.2)

Use ternary operator `error = err["detail"] if isinstance(err, dict) and "detail" in err else str(err)` instead of `if`-`else`-block

Replace `if`-`else`-block with `error = err["detail"] if isinstance(err, dict) and "detail" in err else str(err)`

(SIM108)